design/003-interactive-session/003-interactive-session.md

```diff
@@ -0,0 +1,123 @@
+- Start Date: 2021-12-02
+- Status: Proposed
+
+# Run interactive sessions
+
+## Summary
+
+Propose a new sub-group command `renku session` that is responsible for
+managing interactive sessions, both local and remote, for a project.
+
+## Motivation
+
+Renkulab.io provides the option to run interactive sessions of a project. Among other things the session
+contains a jupyter server.
+
+The motivation behind this RFC is to provide a simple renku CLI sub-command that basically allows the
+starting and stopping of these interactive sessions.
+
+## Design Detail
+
+The proposed sub-commands of `renku session`, namely to run interactive sessions. The local session heavily relies on
+[docker](https://www.docker.com/). Docker is one of the most popular container engines, that eases the development
+and deployment of applications, by off-loading the installation of the runtime environment dependencies of the
+application.
+
+In order to start a local interactive session, just as in case of renkulab.io, it is required to provide
+a `Dockerfile`. This is by default provided by `renku init` when creating a renku project, but if the user
+is using a custom project template it might not be present.
+
+### Proposed Commands
+
+#### renku session start
+
+A command to start an interactive session.
+
+By default, the `Dockerfile` present in the given project is going to be used for building the docker image
+that's going to be used for running the interactive session. In case the docker image is not available the
+user is going to be prompted whether the user wants to build the docker image, as in some cases it can be
+time consuming.
+
+With the `--provider <provider_type>` flag a user can specify session provider to be used for starting
+an interactive session. By default two different providers are shipped with Renku CLI:
+ - docker
+ - renkulab
+The provider specific configuration values can be specified by using the `--config <config.yaml>` flag.
+
+##### Detailed Parameter Description
+
+```
+renku session start [OPTIONS]
+
+--image <image_name>			Override the docker image to be used for the interactive session.
+-p|--provider [docker|renkulab] 	Use the specified backend for starting an interactive session.
+-c|--config   				YAML file containing configuration for the provider.
+--cpu <num of cpus>			Specify how much of the available CPU resources a container can use.
+--disk <disk space>			Specify the amount of disk space for the container.
+--gpu <device>				Specify the GPU allocated for the container.
+--memory <max memory>			The maximum amount of memory the container can use.
```

Comment:
in case of docker, what are the defaults for cpu/memory? Do we take them from the project's configuration if it exists? If it does not exist, do we leave it blank (i.e. use all available resources)?

Reply:
it's the default of the provider...

Reply:
namely i dont see neither the requirement nor the point of trying to unify it in this level... each provider has it's own default values and they are going to be used unless these are set by the user...

Reply:
right, but the user thinks they have set the default by setting it in the project's config

Reply:
it'll be used if it's set in the project's config... unless if it's not set. or maybe i'm misunderstanding here something...?

Reply:
No that's fine, but it wasn't stated so I'm just making sure that's the intent.

Reply:
ah, i'll then just add a sentence in the command description that the resource related options are there to override the [project level resource settings](https://renku.readthedocs.io/en/latest/reference/templates.html?highlight=cpu_request#renku)